pkg/ezkube/resource_id.go

```diff
@@ -88,18 +95,26 @@ func ConvertRefToId(ref deprecatedClusterResourceId) ClusterResourceId {
 	}
 }
 
-func GetDeprecatedClusterName(id ResourceId) string {
-	if id, ok := id.(deprecatedClusterResourceId); ok {
-		return id.GetClusterName()
+func getDeprecatedClusterName(id ResourceId) string {
+	if depResourceId, ok := id.(deprecatedClusterResourceId); ok {
+		return depResourceId.GetClusterName()
+	} else if depZZZResourceId, ok := id.(deprecatedZZZClusterResourceId); ok {
+		return depZZZResourceId.GetZZZ_DeprecatedClusterName()
 	}
 	return ""
 }
 
 func GetClusterName(id ClusterResourceId) string {
-	if id.GetAnnotations() == nil {
-		return ""
+	annotations := id.GetAnnotations()
+	deprecatedClusterName := getDeprecatedClusterName(id)
+
+	if annotations == nil {
+		return deprecatedClusterName
+	} else if annotations[ClusterAnnotation] == "" {
+		return deprecatedClusterName
 	}
```

Comment:
```suggestion
	if annotations == nil || annotations[ClusterAnnotation] == "" {
		return getDeprecatedClusterName(id)
	}
```
nit: only call the deprecated func if needed

Reply:
i like it!